pkg/action/upgrade.go

```diff
@@ -0,0 +1,43 @@
+package action
+
+import (
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"strings"
+)
+
+type Upgrade struct {
+	*action.Upgrade
+	cfg         *Configuration
+	ReleaseName string
+}
+
+// NewUpgrade creates a new Upgrade object with the given configuration.
+func NewUpgrade(cfg *Configuration) *Upgrade {
+	return &Upgrade{
+		Upgrade: action.NewUpgrade(cfg.Configuration),
+		cfg:     cfg,
+	}
+}
+
+func (i *Upgrade) Name(chart *chart.Chart, args []string) (string, error) {
```

Comment:
Missing doc comment on Name(), as it is exported (shouldn't `golangci-lint` catch this? It hasn't on CI Oo).

Silly question, can we just use `pkg/action/install.go.Name()` somehow? https://github.com/rancher-sandbox/hypper/blob/main/pkg/action/install.go#L83-L140

If not, this should be a static function, not a method, as I don't see it making use of the action in  `i`. If it were a static function, I would put it in a separate file in pkg/action/. But it would be nice if we didn't have this logic duplicated.

Reply:
> Silly question, can we just use pkg/action/install.go.Name() somehow? https://github.com/rancher-sandbox/hypper/blob/main/pkg/action/install.go#L83-L140

This would need us instantiating the Install with all the params which is cumbersome and could hide some issues.

> If not, this should be a static function, not a method, as I don't see it making use of the action in i. If it were a static function, I would put it in a separate file in pkg/action/. But it would be nice if we didn't have this logic duplicated.

So I thought about this and Im conflicted.

On one hand, I think we should move this into a `chartutils` package as its common to both
On the other, we may need to do different things to obtain the name in the `Install` vs the `Upgrade`

I lean more heavily into moving it to chartutils, but would like more feedback.

Reply:
:thinking: mmh, true, good point!.
So, both Install and Upgrade are of type Action, so if we need access to Namespace and ReleaseName, both will have it in case we need it in the future (we already need them in `Install.Name()`).

How about creating a new unexported function in pkg/action/name.go (or other), containing the logic on https://github.com/rancher-sandbox/hypper/blob/main/pkg/action/install.go#L108-L115, that both `pkg/action/install.Name()` and `pkg/action/Upgrade.Name()` call?
I would prefer if that new unexported function contained the following logic: https://github.com/rancher-sandbox/hypper/blob/main/pkg/action/install.go#L100-L115 but the amount of flags is different between Install and Upgrade.

Reply:
Do you mind if we do this change afterwards? That way I can get this out of the way and experiment with it later.

Reply:
Ok, let's do that. But then, let's open an issue to deduplicate the Name() code, or I for sure will forget :).